sys/netinet/cc/cc_cdg.c

```diff
@@ -519,8 +520,8 @@ prob_backoff(long qtrend)
 			idx = qtrend;
 
 		/* Backoff probability proportional to rate of queue growth. */
-		p = (INT_MAX / (1 << EXP_PREC)) * probexp[idx];
-		backoff = (random() < p);
+		p = (UINT_MAX / (1 << EXP_PREC)) * probexp[idx];
+		backoff = (prng32() < p);
```

Comment:
Why p is still int, not uint32_t or uint_t?

Reply:
I do think that's a mistake. Should I do `unit32_t` for it?

Reply:
Yes.  I'd say p should be unsigned for correct comparison, and it should have the same range for the probability to work.  So there should be `uint32_t` and `UINT32_MAX`.

Reply:
Thank you for the feedback! I've adjusted `p` accordingly.

Reply:
Thanks. But why did you leave UINT_MAX in its assignment instead of UINT32_MAX I proposed?

Reply:
My bad! I think I missed that.